fpdf/fpdf.py

```diff
@@ -679,7 +677,7 @@ def add_font(self, family, style="", fname=None, uni=False):
                     if self.diffs[i] == diff:
                         d = i
                         break
-                if d == 0:
+                else:
```

Comment:
Could you maybe use this opportunity to replace the `for`-loop by a call to `any()`?

Reply:
I am not sure how to convert this to `any()`. If it searches for the first matching element (and needs its index), then it should be a `next()`. But that would span over several lines, so this way it is probably easier to follow.

I'll leave it this way (fix the other things though) and please feel free to change it later.

Reply:
@Lucas-C Sorry, I am still returning to this one:

Does this `else` ever get executed?

https://github.com/PyFPDF/fpdf2/blob/9fe0f46c1c11855c07ddccfb022fbd45353053d1/fpdf/fpdf.py#L632-L638

The `for` loop goes from `1` to `len(self.diff) + 1`, so the last iteration has `i = len(self.diff)` and therefore there should be an `IndexError` in the line 633. So if this code runs, it has to `break` way before the loop is over.

What was your idea with the `any()`?

Reply:
Sadly this code is currently not covered by any test: https://codecov.io/gh/PyFPDF/fpdf2/src/master/fpdf/fpdf.py

I opened this issue to add a unit test covering this part of code: https://github.com/PyFPDF/fpdf2/issues/92

At the time of my previous comment, I just thought that by using `any()` we could avoid a 4-lines for-loop, without digging much more in the code.

Now, I'm sure this part of the code could be improved for readibility, but I'm more concerned about having non-regression tests covering this usage of `add_font`